Counter/src/app/pokemon-detail/pokemon-detail.component.ts

```diff
@@ -63,5 +81,16 @@ export class PokemonDetailComponent implements OnInit {
       )
     ).subscribe();
   }
-
+  delete() {
+    this.pokemonService.removePokemon(this.id).subscribe(_ => this.location.back());
```

Comment:
You need to manage the lifecycle of this subscription most likely with a `take(1)`.
You also don't want to use `this.location.back()`. Remember that this is a single page application and using back or refreshing the page or anything with location can cause a reload of the entire application or potentially show stale data when we go back. We want to be in control of our data and lifecycle so it's more common to do an angular routing redirect here.

Reply:
Whoops, didn't realize I forgot a take there. I also removed location and changed it to a router usage.